### DISCUSSION TITLE
Extend admin: populate select options based upon entity properties

### DISCUSSION THREAD

**Discussion by fnagel** [sulu/sulu#7668] [Q&A] [answered]
I've integrated custom entities as described here: https://docs.sulu.io/en/2.5/book/extend-admin.html

I'm having a custom entity with a _source_ property, an _attribute_ property and a _value_ property. All are selects. The first two properties (_source_ and _attribute_) determine which options are available in the _value_ select. I have some ideas on how to do that: 

I was not able to find a way to do this using https://docs.sulu.io/en/2.5/cookbook/select-values-service.html as it seems there is no way o pass any other arguments other than a hard-coded string or the _locale_ variable. Is that true?

Another idea was to return a (virtual) property containing an array of options in the entity JSON, so I can use them within the form but that seems not possible too. 

Last but not least: add lots of `<property type="single_select">` with `visibleCondition` directives for each _source_ and _attribute_ combination. Not really a straight forward approach and lots of duplicate code. 

Anyway, this seems to be the only working idea or do I miss something?

**Comment by alexander-schranz** [comment, marked as answer]
The `single_select` and `select values services` are only for hardcoded values and shoult not be used for anything else which I wanted to make clear in the docs with:

> Be aware that the provided expression is only evaluated during the initial request to the administration interface. If you want to provide a selection for your custom entity, you should configure the selection or single_selection field-type as described in [Extend Admin UI](https://docs.sulu.io/en/2.5/book/extend-admin.html).

In my personally opinion I even would remove the `expression` again as they are widely misused. if it is something more as providing PHP or Enum avoid the `expression` and go over selections.

For all things which have todo with entities `selection` and `single_selection` over there APIs are the correct way you can forward query parameters and conditions to return so only data you want and all is 100% in your hand how you handle that query parameters in your custom controllers.

Here some complex example out of a project:

```xml
        <property name="directories" type="directory_selection" multilingual="false">
            <meta>
                <title>sulu_directory.directories</title>
            </meta>
            <params>
                <param name="item_disabled_condition" value="(_permissions &amp;&amp; !_permissions.assign)" />
                <param name="allow_deselect_for_disabled_items" value="false" />
                <param name="request_parameters" type="collection">
                    <param name="includeAssignPermissions" value="true" />
                </param>
                <param name="resource_store_properties_to_request" type="collection">
                    <param name="assignedTopicIds" value="topics" />
                </param>
            </params>
        </property>

        <property name="topics" type="topic_selection" multilingual="false">
            <meta>
                <title>sulu_directory.topics</title>
            </meta>
            <params>
                <param name="item_disabled_condition" value="(_permissions &amp;&amp; !_permissions.assign)" />
                <param name="allow_deselect_for_disabled_items" value="false" />
                <param name="request_parameters" type="collection">
                    <param name="includeAssignPermissions" value="true" />
                </param>
                <param name="resource_store_properties_to_request" type="collection">
                    <param name="assignedDirectoryIds" value="directories" />
                </param>
            </params>
        </property>
```

**Reply by fnagel** [reply]
Thanks for you answer Alex. I will try this as soon as possible and give more feedback on how I tackled this.
